### ISSUE TITLE
Clarification on VSP “reflected” term and cycle-consistency approximation (cc_* losses)

### ISSUE THREAD

**Issue by inoue0426** [rjha18/vec2vec#18]
Thank you for the great paper and code release!
While aligning the implementation with the paper, I noticed two components that seem to extend beyond the original loss definitions. I’d appreciate clarification on whether these are intentional design choices or post-publication improvements.

---

### 1️⃣ VSP “reflected” term

```python
out_sims_reflected = A @ B.T
vsp_loss_reflected = (in_sims - out_sims_reflected).abs().mean()
```

- The paper’s Vector Space Preservation loss only mentions preserving pairwise geometry within each space.
- The code also compares `A @ B.T`, i.e., cross-space similarity between translated and original embeddings.

**Questions**

- Is this “reflected” term an intentional heuristic to better align cross-space structure?
- Was it used in the reported paper results, or is it an implementation extension?

### 2️⃣ Cycle-consistency approximation

```python
cc_ins[out_flag] = translations[out_flag][in_flag].detach()
cc_recons, cc_translations = translator(cc_ins)
cc_rec_loss = rec_loss_fn(...)
cc_trans_loss = trans_loss_fn(...)
cc_vsp_loss = vsp_loss_fn(...)
```

- The paper defines `L_CC(F₂(F₁(x)), x)` explicitly.
- The code instead detaches translated embeddings and reuses them with rec/trans/vsp losses — a stochastic approximation.

**Questions**

- Is this block intended as a practical surrogate for L_CC?
- Which coefficients (loss_coefficient_cc_*) should be used to reproduce the paper baseline?

Thanks

**Comment by rjha18** [comment]
Hi @inoue0426, so sorry for the super late response here! The code in this repo is the code that we used to generate the results in the paper! Answering your other questions:

**VSP:** Yes and yes! The reflected term is sort of a heuristic that we found was helpful when training vec2vecs and is included in the results for the paper.

**CCs:**
1. Yeah this is an approximation for GPU memory (our GPUs are small :))
2. Use `unsupervised.toml` config has our most common settings. For cell-wise reproduction of our tables, see the configs in our [release](https://github.com/rjha18/vec2vec/releases/tag/v1.0.0). It's worth mentioning that seeds are inconsistent across machine, so you may need to try a few!
